app/models/service_announcement.rb

```diff
@@ -1,2 +1,12 @@
 class ServiceAnnouncement < ApplicationRecord
+  validates :color, inclusion: {
+    in: %w( plain red orange yellow olive green teal blue violet purple pink brown grey black )
+  }
+
+  default_scope { active }
+
+  def self.active
```

Comment:
I think best practice for this sort of thing is to use [a named scope](http://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope) so the filter is chainable. Perhaps:

This part of the query really isn't relevant to the active flag. Perhaps:

```ruby
scope :displayable { where("SQL") }
scope :active { where(active: true) }
default_scope { displayable.active }
```